drivers/hmis_external_apis/app/models/hmis_external_apis/external_forms/form_submission.rb

```diff
@@ -34,12 +35,16 @@ def self.from_raw_data(raw_data, object_key:, last_modified:, form_definition:,
         submitted_at: last_modified,
         definition_id: form_definition.id,
         raw_data: raw_data,
+        cleaned_values: clean(raw_data),
       }
       submission.save!
-      submission.process_custom_data_elements!(form_definition: form_definition)
       submission
     end
 
+    protected def clean(raw_data)
+      raw_data.select { |link_id, _| definition.link_id_item_hash.key?(link_id) }
```

Comment:
I think this still needs tweaking, since the values in `raw_data` are saved by custom field key, not by link ID.

Reply:
Hrm, yeah. The form processor will [raise an error](https://github.com/greenriver/hmis-warehouse/blob/6141755f2fd4f0b7f9cc084ca1ce6aa5c9350d6d/drivers/hmis/app/models/hmis/form/form_processor.rb#L78) if it encounters any keys that don't match `mapping`s in the Form Definition, and that's basically the logic we would need to replicate here.  I'm hesitant about re-implementing that logic. Let's discuss

Reply:
I also thought you and Theron had decided the cleaned_values wasn't strictly necessary, and it would be safe to use the raw values for processing and display. 
For processing, we would just need to drop known extraneous keys `captcha_score` ,`form_definition_id`,and `form_content_digest` from the raw_values prior to processing.